src/quacc/recipes/espresso/phonons.py

```diff
@@ -84,12 +84,201 @@ def phonon_job(
     )
 
 
+@job
+def q2r_job(
+    prev_dir: SourceDirectory,
+    fildyn: str = "matdyn",
+    parallel_info: dict[str] | None = None,
+    **calc_kwargs,
+) -> RunSchema:
+    """
+    Function to carry out a basic q2r.x calculation. It should allow you to
+    use all the features of the [q2r.x binary](https://www.quantum-espresso.org/Doc/INPUT_Q2R.html#idm51)
+
+    This job requires the results of a previous ph.x calculation, you might
+    want to create your own flow to run both jobs in sequence.
+
+    Parameters
+    ----------
+    prev_dir
+        Outdir of the previously ran ph.x calculation. This is used to copy
+        the the dynamical matrix files.
+    fildyn
+        The name of the file containing the dynamical matrix. Defaults to "matdyn".
+        If you changed it in the ph.x calculation, you should change it here too.
+    parallel_info
+        Dictionary containing information about the parallelization of the
+        calculation. See the ASE documentation for more information.
+    **calc_kwargs
+        Additional keyword arguments to pass to the Espresso calculator. Set a value to
+        `quacc.Remove` to remove a pre-existing key entirely. See the docstring of
+        [quacc.calculators.espresso.espresso.Espresso][] for more information.
+
+    Returns
+    -------
+    RunSchema
+        Dictionary of results from [quacc.schemas.ase.summarize_run][].
+        See the type-hint for the data structure.
+    """
+
+    calc_defaults = {"input_data": {"input": {"fildyn": fildyn, "flfrc": "q2r.fc"}}}
+
+    prev_dir = {prev_dir: [f"{fildyn}*"]}
```

Comment:
This is smart. The reason that I like this is that you aren't relying on the user to know what files to provide. You're just asking them for the directory, and then the code takes care of the rest. Everyone knows their prior directory, so it's easy.

I wonder if this kind of logic can/should be implemented in other Espresso `@job`s that are inherently a follow-up. It's probably worth thinking about. Right now, they are just assuming the user provides the right thing.